File: separate_the_digits_in_an_array/stdiaa.java
import java.util.*;

class Solution {
    public int[] separateDigits(int[] nums) {
        List<Integer> answer = new ArrayList<>();

        for (int number : nums) {
            List<Integer> digitsOfCurrentNumber = new ArrayList<>();

            while (number > 0) {
                digitsOfCurrentNumber.add(number % 10);
                number /= 10;
            }

            Collections.reverse(digitsOfCurrentNumber);
            answer.addAll(digitsOfCurrentNumber);
        }

        int[] resultArray = new int[answer.size()];
        for (int i = 0; i < resultArray.length; i++) {
            resultArray[i] = answer.get(i);
        }

        return resultArray;
    }
}

File: separate_the_digits_in_an_array/stdiaa.py
def separate_digits(nums: list[int]) -> list[int]:
    answer: list[int] = []

    for number in nums:
        digits_of_current_number: list[int] = []

        while number > 0:
            digits_of_current_number.append(number % 10)
            number //= 10

        digits_of_current_number.reverse()
        answer.extend(digits_of_current_number)

    return answer


print(separate_digits([13, 25, 83, 77]))
print(separate_digits([7, 1, 3, 9]))
